docs/RationaleSeq.md

```diff
@@ -0,0 +1,70 @@
+# Seq(uential) Dialect Rationale
+
+This document describes various design points of the `seq` dialect, why it is
+the way it is, and current status. This follows in the spirit of other [MLIR
+Rationale docs](https://mlir.llvm.org/docs/Rationale/).
+
+## Introduction
+
+Digital logic is generally split into two orthogonal categories:
+combinational and sequential. CIRCT contains a `comb` dialect to model the
+basic combinational operations and the (future) `seq` dialect which is
+discussed here. The intention of the `seq` dialect is to provide a set of
+stateful constructs which can be used to model sequential logic, independent
+of the output method (e.g. SystemVerilog).
+
+## Definitions
+
+For the sake of precision, we use the following definitions:
+
+- **Latch:** A memory element which is only sensitive to the levels of its
```

Comment:
Since the world refers to a clocked latch as a latch, would it make sense to call this an 'unclocked latch' for precision and to avoid confusion?

Reply:
I would also suggest to treat D-Latch and D-Flip-Flop similarly in that they have a clock and accept a value into storage. All the other latch types (SR, JK, toggle, etc.) accept a kind-of strobe signal that changes the output value, but not in the sense of "copying in" an input value. Also I think in practice these are very rare and could be represented e.g. as `seq.sr_latch`. But I could see going for `seq.ff` and `seq.latch` to refer to the D-Latch and D-Flip-Flop as useful.